File: JAVA/OBJECT ORIENTED PROGRAMMING/BankApp/src/Account/Bank.java
package Account;

import java.util.ArrayList;

public class Bank {
    private ArrayList<Account> obaAccounts = new ArrayList<>();

    public Account createAccount(String firstName, String lastName, String correctPin) {
        Account obaAccount = new Account(firstName, lastName, correctPin);
        obaAccounts.add(obaAccount);
        return obaAccount;
    }

    public String getCorrectPin(int accountNumber) {
        Account foundAccount = findAccountByAccountNumber(accountNumber);
        return foundAccount.getCorrectPin();
    }

    private Account findAccountByAccountNumber(int accountNumber) {
        return obaAccounts.get(accountNumber - 1);
    }

    public int getBalance(int accountNumber) {
        Account foundAccount = findAccountByAccountNumber(accountNumber);
        return foundAccount.getBalance("9999");
    }

    public void deposit(int accountNumber, int amount) {
        Account foundAccount = findAccountByAccountNumber(accountNumber);
        foundAccount.deposit(amount, "9999");
    }

    public void withdraw(int amount, String pin, int accountNumber) {
        Account foundAccount = findAccountByAccountNumber(accountNumber);
        foundAccount.withdraw(amount, pin);
    }

    public void transfer(int fromAccountNumber, int toAccountNumber, int amount, String pin) {
        Account fromAccount = findAccountByAccountNumber(fromAccountNumber);
        Account toAccount = findAccountByAccountNumber(toAccountNumber);

        fromAccount.withdraw(amount, pin);
        toAccount.deposit(amount, pin);
    }
}

File: PYTHON/OBSWRLD_BANK/bank.py
from OBSWRLD_BANK.account import Account


class Bank:
    def __init__(self):
        self.oba_accounts = []

    def create_account(self, first_name, last_name, correct_pin):
        oba_accounts = Account(first_name, last_name, correct_pin)
        self.oba_accounts.append(oba_accounts)
        return oba_accounts

    def get_correct_pin(self, account_number):
        found_account = self.find_account_by_account_number(account_number)
        return found_account.get_correct_pin()

    def find_account_by_account_number(self, account_number):
        return self.oba_accounts[account_number - 1]

    def get_balance(self, account_number):
        found_account = self.find_account_by_account_number(account_number)
        return found_account.get_balance("1234")

    def deposit(self, amount, account_number):
        found_account = self.find_account_by_account_number(account_number)
        found_account.deposit(amount, "1234")

    def withdraw(self, amount, account_number, pin):
        found_account = self.find_account_by_account_number(account_number)
        found_account.withdraw(amount, pin)

    def transfer(self, from_account_number, to_account_number, amount, pin):
        from_account = self.find_account_by_account_number(from_account_number)
        to_account_number = self.find_account_by_account_number(to_account_number)


        from_account.withdraw(amount, pin)
        to_account_number.deposit(amount, pin)

